excel-export-service/src/index.ts

```diff
@@ -97,58 +100,91 @@ excelService.get("/version", (req: CustomExpressRequest, res: CustomExpressRespo
   );
 });
 
-excelService.get("/download", async (req: CustomExpressRequest, res: CustomExpressResponse) => {
-  if (req.cookies && req.cookies.token) {
-    req.headers.authorization = req.cookies.token;
-  } else if (req.headers.cookie) {
-    const cookies = req.headers.cookie.split("; ");
-    const authToken = cookies.find((cookie) => cookie.startsWith("token="));
-    if (authToken) {
-      req.headers.authorization = `Bearer ${authToken.split("=")[1]}`;
-    }
-  }
-  const token = req.headers.authorization;
-  if (!token) {
-    req.log.error("No authorization token was provided");
-    return res.status(401).send("Please provide authorization token");
-  }
-
-  try {
-    await getApiVersion(axios, token, res.apiBase);
-  } catch (err) {
-    if (!err.response) {
-      logger.error({ err }, "Cannot connect to API service");
-      return res.status(503).send({ message: "Cannot connect to API service" });
+excelService.get(
+  "/download",
+  forwardError(async (req: CustomExpressRequest, res: CustomExpressResponse) => {
```

Comment:
Personally I don't like higher order function here when only one route is using it. I will aceept it, but I'd prefer 

excelService.get(
  "/download",
  async (req: CustomExpressRequest, res: CustomExpressResponse, next: NextFunction) => {
    try {
      ...
     } catch (err) {
      next(err);
    }
  },
);

It's up to you.

Reply:
@SamuelPull I'm open for discussion how to do it. Since this should be something implemented also in other services. For me there are 2 advantages to this approach ones you addopt it.
- It is much quicker with less code. You don't need to duplicate try/catch block in every route
- And you can easily extent the code `forwardError` function in case you need better error handling or any additional functionality.
If you have better way to do it. I'm open to other options.